tests/test_classifiers/test_sparse_sklearn_classifiers.py

```diff
@@ -0,0 +1,156 @@
+import copy
+from typing import Text, Type, Optional
+
+import pytest
+import scipy.sparse
+import numpy as np
+
+from rasa.shared.nlu.training_data.features import Features
+from rasa.shared.nlu.training_data.training_data import TrainingData
+from rasa.shared.nlu.training_data.message import Message
+from rasa.shared.nlu.constants import (
+    TEXT,
+    INTENT,
+    INTENT_NAME_KEY,
+    FEATURE_TYPE_SENTENCE,
+    PREDICTED_CONFIDENCE_KEY,
+)
+
+from rasa_nlu_examples.classifiers.sparse_naive_bayes_intent_classifier import (
+    SparseNaiveBayesIntentClassifier,
+)
+from rasa_nlu_examples.classifiers.sparse_logistic_regression_intent_classifier import (
+    SparseLogisticRegressionIntentClassifier,
+)
+from rasa_nlu_examples.classifiers.sparse_sklearn_intent_classifier import (
+    SparseSklearnIntentClassifier,
+)
+
+SPARSE_SKLEARN_INTENT_CLASSIFIERS = [
+    SparseNaiveBayesIntentClassifier,
+    SparseLogisticRegressionIntentClassifier,
+]
+
+
+def generate_dummy_training_data(
+    num_messages: int = 10,
+    with_dense_features: bool = False,
+    seed: Optional[int] = None,
+) -> TrainingData:
+    """creates some minimal dummy TrainingData that contains intent training
+    examples with random features, which are sparse by default.
+
+    :param with_dense_features: if set to True, some dense features will be added
+    :param num_messages: number of training_examples to be created
+    :param seed: seed for the rng used to create random features
+    """
+    rng = np.random.default_rng(seed=seed)
+
+    def random_feature(sparse: bool = True):
+        arr = rng.integers(3, size=(10))
+        return Features(
+            features=(scipy.sparse.coo_matrix(arr) if sparse else arr),
+            feature_type=FEATURE_TYPE_SENTENCE,
+            origin="",
+            attribute=TEXT,
+        )
+
+    training_examples = [
+        Message(
+            data={
+                TEXT: "text",  # needed, otherwise not recognised as NLU example
+                INTENT: intent,
+            },
+            features=[random_feature(sparse=True)]
+            + ([random_feature(sparse=False)] if with_dense_features else []),
+        )
+        for intent in ["intent1", "intent2"]
+        for _ in range(num_messages)
+    ]
+    return TrainingData(training_examples=training_examples)
+
```

Comment:
I really like the way you've set up the tests here. My only comment would be that it might be good to add a test for the `intent_ranking`.

Reply:
Oh yep, I've overlooked that. There's probably also a constant for "intent_ranking" that I should use :D

Reply:
added a check that tests if intent_ranking exists (and for the sorting seprately)